jib-gradle-plugin/scripts/prepare_release.sh

```diff
@@ -20,20 +16,24 @@ Die() {
 }
 
 DieUsage() {
-    Die "Usage: ./scripts/prepare_release.sh <release version>"
+    Die "Usage: ./scripts/prepare_release.sh <release version> [<post-release version>]"
 }
 
 # Usage: CheckVersion <version>
 CheckVersion() {
-    [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || Die "Version not in ###.###.### format."
+    [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z]+)?$ ]] || Die "Version: $1 not in ###.###.###[-XXX] format."
 }
 
-[ $# -ne 2 ] || DieUsage
+[ $# -ne 1 ] && [ $# -ne 2 ] && DieUsage
 
 EchoGreen '===== RELEASE SETUP SCRIPT ====='
 
 VERSION=$1
 CheckVersion ${VERSION}
+if [ $2 ]; then
```

Comment:
I think the script will fail if $2 is not given. Try `if [ -n "$2" ];`.

Reply:
This sets `POST_RELEASE_VERSION` if it is given in the command line `[ $2 ];`, and doesn't set it if not.

Reply:
I meant the script will give you a syntax error if $2 is not given or empty, as it ends up `if [ ]`. You need to guard it with quotes.

Reply:
I just realized `[ ]` or `[ -n ]` does not crash, but it's desirable to quote it, as `[ == "something" ]` surely fails: https://stackoverflow.com/questions/53470679/bash-if-statement-with-prepended-character